### DISCUSSION TITLE
Flashforge Dreamer BL touch probe integration

### DISCUSSION THREAD

**Discussion by Denso1980** [moonglow/FlashForge_Marlin#65] [Ideas]
Hi moonglow

I would like to gratulate your awesome project and created a firmware flashforge dreamer 3dprinter.Is it any chance to add a BL touch probe to the machine and auto bed levelling option in your firmware? If it is possible with your firmware can you give me information about this upgrade(connection diagram), please?

best regards
Denes

**Comment by moonglow** [comment]
Hi ! thank you !

Yep it is possible to connect ABL sensor to our board ( i already bought one ) but i need time to check it all and we also need some kind of custom ABL holder attached to our printhead.

**Comment by Denso1980** [comment]
Thanks your reply.I hope will be hearing from you soon. Regards Denes

**Comment by snotten** [comment]
Hello! First off, big thanks for all your work! Just bought a Dreamer, so this is yaay!

Second; are there any pre-compiled firmware files with "Marlin: Uncomment #define HOST_ACTION_COMMANDS in Configuration_adv.h"?

I'm reading about compiling on my own, but I dont wanna brick something ^^

Live long and prosper!

**Comment by MetalDracon** [comment]
Hello, I am currently working on making a BL touch mount for a Flashforge Dreamer and am preparing to switch to it. However as I look at the set up guides on a BL touch I see that a servo pin, and a 5V VCC, and wanted to determine which pins are the best options for these pins. Attached is a picture of my board, for identification purposes.![DreamerBoard](https://user-images.githubusercontent.com/104755011/183567038-39234c88-f353-4d68-8b98-35c865fadf12.jpg)

**Reply by moonglow** [reply]
Hi @MetalDracon !

I think you need to remove your WI-FI module ( my marlin port does not support wifi anyway ) and you will get alot of free I/O pins to add such functionality.  Please refer[ to WIFI socket pinout](https://github.com/moonglow/FlashForge_Marlin/wiki/FlashForge-WIFI-module-pinout) you need a *NEW* 12 pins wifi socket pinout. Use multimeter or something like that to be sure what you connect to right pins ( as example use GND pins as reference ). I do not remember if we have 5V source on our motherboard. I want to find some spare time to write a guide how to enable such functionality, but i really lack of it for now ( thats why i also do not publish new firmware release, because i need to test all my new bugfixes and so on ). If you got success with that i will accept any new guides pulls !

**Reply by MetalDracon** [reply]
Hi @moonglow 
I went and poked at my motherboard some more, looking at all the unused connections after removing the WIFI module. I found no active 5V port, but did find one that was label 5V, it is the camera port hiding behind some wires in my previous picture, but it was not energized. In addition, I did find a test spot for the 5V line on the board and did find it to be correctly calibrated. 
Lastly, I found one of the pins on the EDetect output to be a 24V pin, the second pin from the left in the previous picture. However, the far right pin was 3.3V.  
The lack of 5V is not dissuading me from continuing because I am running an octoprint sever so I can draw power from that.

**Reply by moonglow** [reply]
@MetalDracon 
>camera port

you can enable/disable power on it via CAMERA_PWR_ON_PIN pin.

for old motherboard ( or as universal solution ) we can use external little dc/dc to convert 24V to 5V.

p.s: all pins what i already found described [HERE](https://github.com/moonglow/FlashForge_Marlin/blob/marlin_2.0.x/Marlin/Marlin/src/pins/stm32f4/pins_FF_MOTHERBOARD.h)
